lib/workers/repository/config-migration/branch/index.ts

```diff
@@ -0,0 +1,49 @@
+import { GlobalConfig } from '../../../../config/global';
+import type { RenovateConfig } from '../../../../config/types';
+import { logger } from '../../../../logger';
+import { platform } from '../../../../modules/platform';
+import { checkoutBranch } from '../../../../util/git';
+import { getMigrationBranchName } from '../common';
+import { createConfigMigrationBranch } from './create';
+import { MigratedDataFactory } from './migrated-data';
+import { rebaseMigrationBranch } from './rebase';
+
+export async function checkConfigMigrationBranch(
+  config: RenovateConfig
+): Promise<string | null> {
+  logger.debug('checkConfigMigrationBranch()');
+  const migratedConfigData = await MigratedDataFactory.getAsync();
+  if (!migratedConfigData) {
+    logger.debug(
+      'checkConfigMigrationBranch() Config does not need migration\n'
+    );
+    return null;
+  }
+  const configMigrationBranch = getMigrationBranchName(config);
+  if (await migrationPrExists(configMigrationBranch)) {
+    logger.debug('Config Migration PR already exists');
+    await rebaseMigrationBranch(config, migratedConfigData);
+
+    if (platform.refreshPr) {
+      const configMigrationPr = await platform.getBranchPr(
+        configMigrationBranch
+      );
+      if (configMigrationPr) {
+        await platform.refreshPr(configMigrationPr.number);
+      }
+    }
+  } else {
+    logger.debug('Config Migration PR does not exist');
+    logger.debug('Need to create migration PR');
+    await createConfigMigrationBranch(config, migratedConfigData);
+  }
+  if (!GlobalConfig.get('dryRun')) {
```

Comment:
Can you please test this in real for the new different vbalues for `dryRun`?

https://docs.renovatebot.com/self-hosted-configuration/#dryrun

Reply:
`full`: 
```
 INFO: DRY-RUN: Would commit files to config migration branch (repository=ladzaretti/cfg-testing)
 INFO: DRY-RUN: Would create migration PR (repository=ladzaretti/cfg-testing)
```
no actual PR created.

`lookup` | `extract`: does not get to execute related code.

`null`: behaves as expected, opens a PR